### PR TITLE
Arcmining QoL: GPS component to scanned vents and vent(pin)pointer

### PR DESCRIPTION
**Pull request by Iajret**
## About The Pull Request
Scanning ore vents will now tag them on GPS so you can find them easily when you are prepared (and remove them when they are tapped). Also adds special pinpointer to help with finding unscanned vents in a first place. And adds missing uranium overlays for scanned vents. Because why not.
<details>

image

image
</details>

## Why It's Good For The Game
While looking at mineral density to find vents is somewhat fun and entertaining, having some sort of upgrade to make finding them easier adds a bit of a progression. Also re-finding already scanned vents can be a bit annoying. You need to fill your backpacks with GPSs and tag them manually. It is also inconsistent with geysers that are added to positioning after you scan them. And with tendrils, which are always visible.
## Changelog
:cl:
add: added ventpointer that points toward nearby ore vents and can be bought with mining points
balance: scanned ore vents can now be found with GPS
fix: fixed ore vents missing overlay icon for uranium
/:cl:

### CODE REVIEW

**Comment by iamcooldan** [comment]
does the pinpointer only show the nearest vent, does it search for and show the nearest untapped vent, or is there a possibility of both and switching between modes?

**Comment by iamcooldan** [comment]
lovely

**Comment by Stoopler** [comment]
what's the chances that a fix for (i admittedly haven't searched to see if its even been reported) the exosuit mining scanner's "scan vents" not working could be thrown in this :D

**Comment by newfren** [comment]
is there a way to tell on the gps if the vent has been successfully cleared or not post tapping?

**Comment by Wallemations** [comment]
I could have sworn we already had a PR merged for automatically adding GPS signals to scanned vents

**Comment by Iajret** [comment]
> what's the chances that a fix for (i admittedly haven't searched to see if its even been reported) the exosuit mining scanner's "scan vents" not working could be thrown in this :D

i can look at it, but im not sure if piling several random mining related things in to one PR is good thing

> is there a way to tell on the gps if the vent has been successfully cleared or not post tapping?

Ye, i have forgotten to add about this in PR body. Tapping vents will remove them from GPS list to reduce clutter. Also, you can't identify ores in vents by using GPS. While it could be added, i don't think it is worth the trouble. Call my laziness a feature. A feature to make miners not so casually easy.

> I could have sworn we already had a PR merged for automatically adding GPS signals to scanned vents

It is something that could've been added long time ago. But to my surprise, i wasn't able to find even closed PRs.


reeeeeee, reviews where

File: code/game/objects/structures/lavaland/ore_vent.dm [hunk 1]
```diff
@@ -270,6 +270,7 @@
 		balloon_alert_to_viewers("vent tapped!")
 		icon_state = icon_state_tapped
 		update_appearance(UPDATE_ICON_STATE)
+		qdel(GetComponent(/datum/component/gps))
```
Comment (ArcaneMusic) [inline code/game/objects/structures/lavaland/ore_vent.dm:273]: If you want to go for bonus points, we may want to make a signal that the GPS component can handle it's own deletion, but for now I think this is acceptable 👍

**Review by ArcaneMusic — APPROVED** [review]
Looks clean, nice work

**Pull request merged**